src/sentry_plugins/vsts/client.py

```diff
@@ -1,29 +1,36 @@
 from __future__ import absolute_import
 
+import six
+import sys
+
 from sentry_plugins.client import AuthApiClient
-from sentry_plugins.exceptions import UnsupportedResponseType
+from sentry_plugins.exceptions import ApiUnauthorized, UnsupportedResponseType
 
 
 class VstsClient(AuthApiClient):
     api_version = '3.0'
 
-    def exception_means_unauthorized(self, exc):
-        rv = super(VstsClient, self).exception_means_unauthorized(exc)
-        if rv:
-            return rv
-        # VS seems to return an HTML page with an odd status code when our token
-        # is invalid
-        elif isinstance(exc, UnsupportedResponseType) and exc.code == 203:
-            return True
-        return False
-
     def request(self, method, path, data=None, params=None):
         headers = {
             'Accept': 'application/json; api-version={}'.format(self.api_version),
             'Content-Type': 'application/json',
             'X-HTTP-Method-Override': method,
         }
-        return self._request(method, path, headers=headers, data=data, params=params)
+        try:
+            return self._request(method, path, headers=headers, data=data, params=params)
+        except UnsupportedResponseType as exc:
+            # XXX(dcramer): I'm not entirely convinced this is an authentication
+            # error, but we've seen in the past that refreshing the token would
+            # resolve this response, and unfortunately there's not much detail
+            # online about it.
+            # https://github.com/getsentry/sentry/issues/5263
+            if exc.code == 203:
+                six.reraise(
+                    ApiUnauthorized,
+                    ApiUnauthorized('Received an HTTP %s %s response from the API.', exc.code),
+                    sys.exc_info[2],
```

Comment:
TIL. does this need to be `sys.exc_info()[2]`?

Reply:
good catch